app/mailers/discovered_mailer.rb

```diff
@@ -1,7 +1,18 @@
 class DiscoveredMailer < ::ApplicationMailer
   helper :discovered_hosts
   def discovered_summary(options = {})
-    raise ::Foreman::Exception.new(N_("Must specify a user with email enabled")) unless (user = User.find(options[:user])) && user.mail_enabled?
+    user = if options[:user].kind_of? User
+             options[:user]
+           elsif options[:user].kind_of? Integer
+             User.find(options[:user])
+           else
+             raise ::Foreman::Exception.new(N_('Invalid user type of %s was provided') % options[:user].class.to_s)
```

Comment:
I still see `%` instead `,` this is not correct. Exception constructor requires all arguments to be passed via arguments.

Reply:
From https://projects.theforeman.org/projects/foreman/wiki/Translating#Exceptions

We have defined two exceptions `Foreman::Exception` and `Foreman::WrappedException`. Both are generic exceptions which are i18n-friendly. Use the following style to properly format messages.

Let me show this to you:

```
[4] pry(main)> ::Foreman::Exception.new(N_('Invalid user type of %s was provided') % 'DummyType').code
=> "ERF42-7023"
[5] pry(main)> ::Foreman::Exception.new(N_('Invalid user type of %s was provided') % 'AnotherType').code
=> "ERF42-5230"
[6] pry(main)> ::Foreman::Exception.new(N_('Invalid user type of %s was provided'), 'DummyType').code
=> "ERF42-9840"
[7] pry(main)> ::Foreman::Exception.new(N_('Invalid user type of %s was provided'), 'AnotherType').code
=> "ERF42-9840"
```

Reply:
oh, now i get it, didn't understand it without your explanation :/ 
thanks

Reply:
Sorry my bad, it's just one character, very hidden indeed :)